tests/functional/services/flows/test_flow_crud.py

```diff
@@ -23,12 +21,34 @@ def test_create_flow(flows_client, subscription_id):
         assert "subscription_id" not in req_body
 
 
+@pytest.mark.parametrize("value", [None, [], "dummy_value"])
+@pytest.mark.parametrize("key", ["run_managers", "run_monitors"])
+def test_create_flow_request_data(flows_client, key, value):
+    meta = load_response(flows_client.create_flow).metadata
+
+    # Remove the key or set as empty list
+    if value is None:
+        meta["params"].pop(key)
+    else:
+        meta["params"][key] = value
```

Comment:
`load_response` is pulling global data from a registry.
Therefore, this is modifying a global dict.

We should not modify a loaded `RegisteredResponse.metadata` -- or any other attributes of that global object -- in the course of a test. We can create these constructs in local scope if we need modifiable versions, but the globals should be treated as immutable.

Reply:
@sirosen Ok, I'm hoping I finally got it this time. I registered a new response within the scope of that test. 🤞